src/storage/src/monitor/hitmap.rs

```diff
@@ -0,0 +1,268 @@
+        let data = [(); N].map(|_| AtomicU64::default());
+        let data = Arc::new(data);
+        Self { data }
+    }
+}
+
+impl<const N: usize> Hitmap<N> {
+    pub const fn bits() -> usize {
+        N * u64::BITS as usize
+    }
+
+    pub const fn bytes() -> usize {
+        N * 8
+    }
+
+    pub fn reset(&self) {
+        for elem in &*self.data {
+            elem.store(0, Ordering::Relaxed);
+        }
+    }
+
+    pub fn fill(&self, start_bit: usize, end_bit: usize) {
+        const MASK: usize = (1 << 6) - 1;
+
+        let end_bit = std::cmp::min(end_bit, Self::bits());
+
+        let head_bits = start_bit & MASK;
+        let tail_bits_rev = end_bit & MASK;
+
+        let head_elem = start_bit >> 6;
+        let tail_elem = end_bit >> 6;
+
+        for i in head_elem..=std::cmp::min(tail_elem, N - 1) {
+            let elem = &self.data[i];
+            let mut umask = 0u64;
+            if i == head_elem {
+                umask |= (1u64 << head_bits) - 1;
+            }
+            if i == tail_elem {
+                umask |= !((1u64 << tail_bits_rev) - 1);
+            }
+            elem.fetch_or(!umask, Ordering::Relaxed);
+        }
+    }
+
+    pub fn fill_with_ratio(&self, start: f64, end: f64) {
+        let start_bit = (Self::bits() as f64 * start) as usize;
+        let end_bit = (Self::bits() as f64 * end) as usize;
+        self.fill(start_bit, end_bit)
+    }
+
+    pub fn local(&self) -> LocalHitmap<N> {
+        LocalHitmap::new(self)
+    }
+
+    pub fn apply(&self, local: &LocalHitmap<N>) {
+        for (global, local) in self.data.iter().zip_eq_fast(local.data.iter()) {
+            global.fetch_add(*local, Ordering::Relaxed);
+        }
+    }
+
+    pub fn ones(&self) -> usize {
+        let mut res = 0;
+        for elem in &*self.data {
+            res += elem.load(Ordering::Relaxed).count_ones() as usize;
+        }
+        res
+    }
+
+    pub fn zeros(&self) -> usize {
+        Self::bits() - self.ones()
+    }
+
+    pub fn ratio(&self) -> f64 {
+        self.ones() as f64 / Self::bits() as f64
+    }
+
+    #[cfg(test)]
+    pub fn to_hex_vec(&self) -> Vec<String> {
+        use itertools::Itertools;
+        self.data
+            .iter()
+            .map(|elem| elem.load(Ordering::Relaxed))
+            .map(|v| format!("{v:016x}"))
+            .collect_vec()
+    }
+}
+
+#[derive(Debug)]
+pub struct LocalHitmap<const N: usize> {
+    owner: Hitmap<N>,
+    data: Box<[u64; N]>,
+}
+
+impl<const N: usize> LocalHitmap<N> {
+    pub const fn bits() -> usize {
+        N * u64::BITS as usize
+    }
+
+    pub const fn bytes() -> usize {
+        N * 8
+    }
+
+    pub fn new(owner: &Hitmap<N>) -> Self {
+        Self {
+            owner: owner.clone(),
+            data: Box::new([0; N]),
+        }
+    }
+
+    pub fn reset(&mut self) {
+        for elem in &mut *self.data {
+            *elem = 0;
+        }
+    }
+
+    pub fn fill(&mut self, start_bit: usize, end_bit: usize) {
```

Comment:
`fill` of `LocalHitmap` and `Hitmap` is basically the same. Is it possible to unify their code in a macro or generic method?

Or do you have any usage of `fill` of `Hitmap`? If you only use the `fill` of `LocalHitmap`, we may only keep the fill of `LocalHitmap`.